src/kv/mod.rs

```diff
@@ -39,6 +59,13 @@ impl Kv {
     /// Will return `Err` if RPC call is failed.
     #[inline]
     pub async fn put(&mut self, req: EtcdPutRequest) -> Res<EtcdPutResponse> {
+        if self.cache_enable {
```

Comment:
Since the cache update and the storage (i.e. etcd server) update is not atomic, it might cause inconsistency between the etcd storage and the cache.
For example, 
```
Time ---------------------------------------------------------------------->
Thread 0 : cache(a, 1)                                         persist(a, 1)
Thread 1 :                     cache(a, 2) persist(a, 2)                      
```
Finally, the storage is `(a, 1)`, while the cache is `(a, 2)`.

Reply:
Yes, currently we can not make the cache update and the storage update atomic, because of many reasons, like the network latency. So we will also add the staleness time for cache in future, then the user will finally get the latest data in some time. Besides, we can also use some mechanism to check the version of the data in cache, if the version of the data in cache is changed when the data comes from server, it means the data maybe obsolete. 

And the scenario here is that the data cache is for read heavy workload, so the probability of data inconsistency should be very minimal.